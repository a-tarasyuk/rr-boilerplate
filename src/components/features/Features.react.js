import './features.scss';
import React from 'react';
import Feature from 'components/feature/Feature.react';

export default ({ features }) => {
  return <div className="features">
    <h1 className="features__title">Boilerplate contains</h1>
    <ul className="features__list">
      { features.map((feature, index) => <Feature key={ index } { ...feature } />) }
    </ul>
  </div>;
};