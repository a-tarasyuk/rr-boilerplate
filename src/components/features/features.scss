.features {
  &__title {
    font-size: 2em;
    margin: 5px 0;
  }

  &__list {
    margin: 0;
    padding: 0;
  }
}