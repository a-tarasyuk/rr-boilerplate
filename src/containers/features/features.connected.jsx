import { connect } from 'react-redux';
import Container from './features';

export const Features = connect(
  ({ features }) => ({ features }),
)(Container);

export default Features;
