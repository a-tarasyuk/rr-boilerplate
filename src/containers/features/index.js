import React from 'react';
import { connect } from 'react-redux';
import { A } from 'components';

// import Features from 'components/features/Features.react';

//const FeaturesContainer= (props) => (
//  <Features {...props} />
//);

//export default connect(
//  ({ features, routing }) => ({ features, routing })
//)(FeaturesContainer);