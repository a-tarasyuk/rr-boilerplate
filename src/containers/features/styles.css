.features {
  margin: 0;
  padding: 0 0 0 20px;
}

.feature {
  margin: 0;
  padding: 10px 5px;
  list-style-type: decimal;
}
