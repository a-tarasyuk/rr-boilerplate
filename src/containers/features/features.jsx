import React from 'react';
import PropTypes from 'prop-types';
import { Link } from '../../components/link';
import styles from './features.css';

export const Features = ({ features }) => {
  features = features.map((feature, index) => (
    <li className={ styles.feature } key={ index }>
      <Link { ...feature } />
    </li>
  ));

  return (
    <ol className={ styles.features }>
      { features }
    </ol>
  );
};

Features.propTypes = {
  features: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Features;
