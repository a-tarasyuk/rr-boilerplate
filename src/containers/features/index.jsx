import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import A from 'components/a';
import styles from './styles.css';


const Features = ({ features }) => {
  features = features.map((feature, index) => (
    <li className={styles.feature} key={index}>
      <A {...feature} />
    </li>
  ));

  return (<ol className={styles.features}>
    {features}
  </ol>);
};

Features.propTypes = {
  features: PropTypes.arrayOf(React.PropTypes.object).isRequired,
};

export default connect(
  ({ features }) => ({ features }),
)(Features);
