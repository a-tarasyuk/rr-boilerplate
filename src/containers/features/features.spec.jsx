import React from 'react';
import { shallow } from 'enzyme';
import { Features } from './features';

const features = [{
  name: 'React',
  href: 'http://facebook.github.io/react',
}];

describe('<Features />', () => {
  it('matches its snapshot', () => {
    const component = shallow(
      <Features features={ features } />
    );

    expect(component).toMatchSnapshot();
  });
});
