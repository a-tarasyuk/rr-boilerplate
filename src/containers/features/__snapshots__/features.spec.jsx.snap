// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Features /> matches its snapshot 1`] = `
<ol
  className="features"
>
  <li
    className="feature"
    key="0"
  >
    <Link
      href="http://facebook.github.io/react"
      name="React"
      title=""
    />
  </li>
</ol>
`;
